src/openvic-simulation/types/Sliders.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+  // TODO: make struct for tariff slider, unlike other sliders, you can have negative values
+  struct slider_value_t {
+  friend struct  SliderManager;
+  private:
+      int PROPERTY(min);
+      int PROPERTY(max);
+      int PROPERTY(value);
+
+  public:
+      constexpr slider_value_t(int value, int min = 0, int max = 100) {
+        // You *can* actually have min > max in Victoria 2
+        // Such a situation will result in only being able to be move between the max and min value. 
+        // This logic replicates this "feature"
```

Comment:
Shouldn't the logic below apply only in the `max < min` case, and not in the normal `min <= max` case?
Also [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) could be useful, particularly for the normal case.
Furthermore, there shouldn't be any unique logic in the constructor here (except maybe for constants like the final min and max variables, once they're added), instead it should set/check values using the regular setter functions so the exact same checks occur when first constructing this object and later when updating it.